ctapipe/reco/impact.py

```diff
@@ -783,232 +963,68 @@ def minimise(self, params, step, limits, minimiser_name="minuit", max_calls=0):
-            opt.set_xtol_rel(1e-3)
-            if max_calls:
-                opt.set_maxeval(max_calls)
-
-            x = opt.optimize(np.asarray(params))
-
-            return x, (0, 0, 0, 0, 0, 0), self.get_likelihood_min(x)
-
-        elif minimiser_name in ("lm", "trf", "dogleg"):
-            self.array_return = True
-
-            min = least_squares(
-                self.get_likelihood_min,
-                params,
-                method=minimiser_name,
-                x_scale=step,
-                xtol=1e-10,
-                ftol=1e-10,
-            )
-
-            return min.x, (0, 0, 0, 0, 0, 0), self.get_likelihood_min(min.x)
-
-        else:
-            min = minimize(
-                self.get_likelihood_min,
-                np.array(params),
-                method=minimiser_name,
-                bounds=limits,
-                options={"disp": False},
-                tol=1e-5,
-            )
-
-            return np.array(min.x), (0, 0, 0, 0, 0, 0), self.get_likelihood_min(min.x)
-
 
-def spread_line_seed(
-    hillas,
-    tel_x,
-    tel_y,
-    source_x,
-    source_y,
-    tilt_x,
-    tilt_y,
-    energy,
-    shift_frac=[2, 1.5, 1, 0.5, 0, -0.5, -1, -1.5],
-):
-    """
-    Parameters
-    ----------
-    hillas: list
-        Hillas parameters in event
-    tel_x: list
-        telescope X positions in tilted system
-    tel_y: list
-        telescope Y positions in tilted system
-    source_x: float
-        Source X position in nominal system (radians)
-    source_y:float
-        Source Y position in nominal system (radians)
-    tilt_x: float
-        Core X position in tilited system (radians)
-    tilt_y: float
-        Core Y position in tilited system (radians)
-    energy: float
-        Energy in TeV
-    shift_frac: list
-        Fractional values to shist source and core positions
-
-    Returns
-    -------
-    list of seed positions to try
-    """
-    centre_x, centre_y, amp = list(), list(), list()
-
-    for tel_hillas in hillas:
-        centre_x.append(tel_hillas.x.to(u.rad).value)
-        centre_y.append(tel_hillas.y.to(u.rad).value)
-        amp.append(tel_hillas.intensity)
-
-    centre_x = np.average(centre_x, weights=amp)
-    centre_y = np.average(centre_y, weights=amp)
-    centre_tel_x = np.average(tel_x, weights=amp)
-    centre_tel_y = np.average(tel_y, weights=amp)
-
-    diff_x = source_x - centre_x
-    diff_y = source_y - centre_y
-    diff_tel_x = tilt_x - centre_tel_x
-    diff_tel_y = tilt_y - centre_tel_y
-
-    seed_list = list()
-
-    for shift in shift_frac:
-        seed_list.append(
-            create_seed(
-                centre_x + (diff_x * shift),
-                centre_y + (diff_y * shift),
-                centre_tel_x + (diff_tel_x * shift),
-                centre_tel_y + (diff_tel_y * shift),
-                energy,
-            )
+        energy = params[4]
+        xmax_scale = 1
+
+        # Now do the minimisation proper
+        self.min = Minuit(
```

Comment:
rename `self.min` → `self.minimizer`  to be more clear

Reply:
Why assign to self at all?

Reply:
good point, it's not used elsewhere. I guess it was done that way for interactive debugging purposes or something, but clearly not needed.